cluster-autoscaler/Dockerfile

```
@@ -11,9 +11,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG BASEIMAGE=gcr.io/distroless/static:nonroot-amd64
-FROM $BASEIMAGE
+FROM --platform=$BUILDPLATFORM golang:1.23 as builder
+
+WORKDIR /workspace
+
+COPY . .
+
+ENV GOARCH=${GOARCH}
+ENV LDFLAGS_FLAG=${LDFLAGS_FLAG}
+ENV TAGS_FLAG=${TAGS_FLAG}
+
+RUN CGO_ENABLED=0 GOOS=linux go build -o cluster-autoscaler-$GOARCH $LDFLAGS_FLAG $TAGS_FLAG
+FROM gcr.io/distroless/static:nonroot
+COPY cluster-autoscaler-$GOARCH /cluster-autoscaler
```

```suggestion
COPY --from=builder cluster-autoscaler-$GOARCH /cluster-autoscaler
```

(To make sure it copies the output from the previous step)

---

do we need the ARG GOARCH repeated?

thx for allowing me to avoid a refresher on the dockerfile API! 😝

---

I seem to be getting misleading info, from docs:

- https://docs.docker.com/build/building/variables/#scoping

The above says "Once a build argument is declared or consumed in a stage, it's automatically inherited by child stages." (which suggests we shouldn't have to declare twice.

But this:

- https://docs.docker.com/reference/dockerfile/#scope

says "An ARG variable declared within a build stage is automatically inherited by other stages based on that stage. Unrelated build stages do not have access to the variable. To use an argument in multiple distinct stages, each stage must include the ARG instruction, or they must both be based on a shared base stage in the same Dockerfile where the variable is declared." says that we will have to re-declare ARG if `FROM gcr.io/distroless/static:nonroot` begins a distinct stage and not another stage based on the build stage.

I'm not able to glean from official docs which one or the other of those two multi-stage relationships our 2nd stage is! So I'll just take your suggestion. :)